src/treemap/split.js

```diff
@@ -0,0 +1,105 @@
+export default function(parent, x0, y0, x1, y1) {
+
+  parent.x0 = x0;
+  parent.x1 = x1;
+  parent.y0 = y0;
+  parent.y1 = y1;
+
+  // exit if no children
+  if (!(parent.children) || parent.children.length === 0) {
+    return;
+  }
+
+  splitTree(parent.children, parent.value, x0, y0, x1, y1);
+
+  function splitTree(nodes, sum, x0, y0, x1, y1) {
+
+    var i, n = nodes.length;
+
+    if (n === 0) {
+      return;
+    }
+
+    if (n === 1) {
+      nodes[0].x0 = x0;
+      nodes[0].x1 = x1;
+      nodes[0].y0 = y0;
+      nodes[0].y1 = y1;
+      if (nodes[0].children && nodes[0].children.length > 0) {
+        splitTree(nodes[0].children, parent.value, x0, y0, x1, y1);
+        return;
+      } else {
+        return;
+      }
+    }
+
+    var width = x1 - x0,
+      height = y1 - y0;
+
+    var list1 = [],
+      list2 = [],
+      r1x0 = 0,
+      r1x1 = 0,
+      r1y0 = 0,
+      r1y1 = 0,
+      r2x0 = 0,
+      r2x1 = 0,
+      r2y0 = 0,
+      r2y1 = 0,
+      halfSize = 0,
+      tmp = 0,
+      w1 = 0,
+      w2 = 0;
+
+
+    if (sum === null) {
```

Comment:
I introduced the `sum` argument to `splitTree()` to avoid duplicating the sum calculation.  However, as you said above, there are cases where we might not know the `sum` in which we will call `splitTree(nodes, null, ...)` and expect the `sum` to be calculated.